### PR TITLE
[Platform-Validation] Add Check to Verify Users

### PR DESCRIPTION
**Pull request by nimanch**
This PR Adds a Check to ensure that system interaction with a device to get user information works as intended. The PR Does the following
1. Define a common function for querying user information.
2. Use the common function in iotedge config apply
3. Use the same function in IoT Edge Check to verify users for running IoT Edge are present

Todo: 
1. Add TPM User handling since it is a special scenario

Testing
1. Validated on Ubuntu 20.04 AMD64 , Installed IoT Edge and removed one user

image

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [ ] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [ ] Title of the pull request is clear and informative.
- [ ] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [ ] local testing done.  

### Draft PRs
- Open the PR in `Draft` mode if it is:
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.

_Note: We use the kodiakhq bot to merge PRs once the necessary checks and approvals are in place. When it merges a PR, kodiakhq converts the PR title to the commit title, PR description to the commit description, and squashes all the commits in the PR to a single commit. The net effect is that entire PR becomes a single commit. Please follow the best practices mentioned [here](https://chris.beams.io/posts/git-commit/#:~:text=The%20seven%20rules%20of%20a%20great%20Git%20commit,what%20and%20why%20vs.%20how%20For%20example%3A%20) for the PR title and description_

### CODE REVIEW

File: edgelet/iotedge/src/check/checks/check_users.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            .unwrap_or_else(CheckResult::Failed)
+    }
+}
+
+impl CheckUsers {
+    #[allow(clippy::unused_self)]
+    #[allow(unused_variables)]
+    async fn inner_execute(&mut self, check: &mut Check) -> Result<CheckResult, failure::Error> {
+        // Todo : Add Check for TPM User.
+        for user in ["aziotcs", "aziotks", "aziotid", "iotedge"] {
```
Comment (yophilav) [inline edgelet/iotedge/src/check/checks/check_users.rs:32]: What about `aziottpm` ?

File: edgelet/iotedge/src/config/apply.rs [hunk 2]
```diff
@@ -24,49 +26,16 @@ pub fn execute(config: &Path) -> Result<(), std::borrow::Cow<'static, str>> {
...
-        } else {
-            return Err("this command must be run as root".into());
-        };
+    let (aziotks_user, aziotcs_user, aziotid_user, aziottpm_user, iotedge_user) = {
+        (
+            get_system_user("aziotks").map_err(|err| format!("{}", err))?,
+            get_system_user("aziotcs").map_err(|err| format!("{}", err))?,
+            get_system_user("aziotid").map_err(|err| format!("{}", err))?,
+            get_system_user("aziotpm").map_err(|err| format!("{}", err))?,
+            get_system_user("iotedge").map_err(|err| format!("{}", err))?,
```
Comment (yophilav) [inline edgelet/iotedge/src/config/apply.rs:36]: Nice clean up!

File: edgelet/iotedge/src/check/checks/check_users.rs (hunk 1 above)
Comment (nimanch) [reply edgelet/iotedge/src/check/checks/check_users.rs:32]: I have a todo to address that(See todo in code). TPM User requires some special handling and relies on provisioning information

File: edgelet/iotedge/src/lib.rs [hunk 3]
```diff
@@ -23,6 +23,7 @@ mod check;
 mod client;
 pub mod config;
 mod error;
+mod internal;
```
Comment (ggjjj) [inline edgelet/iotedge/src/lib.rs:26]: why define in internal and not in config folder directly?

File: edgelet/iotedge/src/internal/common.rs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    if Uid::current().is_root() {
+        Ok(User::from_name(name)
+            .with_context(|| format!("could not query {} user information", name))?
+            .ok_or_else(|| anyhow!("could not query {} user information", name))?)
+    } else if cfg!(debug_assertions) {
+        Ok(User::from_uid(Uid::current())
+            .context("could not query current user information")?
+            .ok_or_else(|| anyhow!("could not query current user information"))?)
+    } else {
+        Err(anyhow!("this command must be run as root"))
```
Comment (ggjjj) [inline edgelet/iotedge/src/internal/common.rs:23]: Neat use of anyhow! :)

**Review by ggjjj — COMMENTED** [review]
LGTM

**Review by yophilav — APPROVED** [review]

File: edgelet/iotedge/src/lib.rs (hunk 3 above)
Comment (nimanch) [reply edgelet/iotedge/src/lib.rs:26]: Because we dont want to expose this to outside components.

**Pull request merged**